bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/handler/AccountHandler.java

```diff
@@ -785,12 +786,35 @@ private void handlePushActivity(@Nullable String payload) {
         }
 
         String search = key.registeredUserId + "#" + key.entryId;
-        connection.getActivities(10, pushActivity.timestamp).stream().filter(activity -> search.equals(activity.id))
-                .findFirst()
-                .ifPresent(currentActivity -> currentActivity.getSourceDeviceIds().stream()
-                        .map(sourceDeviceId -> findEchoHandlerBySerialNumber(sourceDeviceId.serialNumber))
-                        .filter(Objects::nonNull).forEach(echoHandler -> Objects.requireNonNull(echoHandler)
-                                .handlePushActivity(currentActivity)));
+
+        /*
+         * DEPRECATED
+         * connection.getActivities(10, pushActivity.timestamp).stream().filter(activity -> search.equals(activity.id))
+         * .findFirst()
+         * .ifPresent(currentActivity -> currentActivity.getSourceDeviceIds().stream()
+         * .map(sourceDeviceId -> findEchoHandlerBySerialNumber(sourceDeviceId.serialNumber))
+         * .filter(Objects::nonNull).forEach(echoHandler -> Objects.requireNonNull(echoHandler)
+         * .handlePushActivity(currentActivity)));
+         */
+
+        Long timestamp = pushActivity.timestamp;
```

Comment:
```suggestion
        long timestamp = pushActivity.timestamp;
```

use primitives where possible

Reply:
timestamp of pushActivity is a nullable Long so have to use Long and check for null